torch/utils/data/datapipes/iter/callable.py

```diff
@@ -82,6 +82,20 @@ def __setstate__(self, state):
             self.fn = dill_function  # type: ignore
 
 
+@functional_datapipe('map')
+class MapIterDataPipe(_CallableIterDataPipe):
```

Comment:
Want to hear your opinion to combine `_CallableIterDataPipe` and `MapIterDataPipe` together and make `FilterIterDataPipe` directly inherited from `IterDataPipe` rather than from `Map` or `_Callable`.

Reply:
Discussed during call: so far leaving it as is.